bin/cliain/src/commands.rs

```diff
@@ -0,0 +1,228 @@
+    VestOther {
+        /// Account seed for which vesting should be performed.
+        #[clap(long)]
+        vesting_account: String,
+    },
+
+    /// Transfer funds via balances pallet
+    VestedTransfer {
+        /// Number of tokens to send.
+        #[clap(long)]
+        amount_in_tokens: u64,
+
+        /// Seed of the target account.
+        #[clap(long)]
+        to_account: String,
+
+        /// How much balance (in rappens, not in tokens) should be unlocked per block.
+        #[clap(long)]
+        per_block: Balance,
+
+        /// Block number when unlocking should start.
+        #[clap(long)]
+        starting_block: BlockNumber,
+    },
+
+    /// Print debug info of storage
+    DebugStorage,
+
+    /// Uploads new code without instantiating a contract from it
+    /// https://polkadot.js.org/docs/substrate/extrinsics/#uploadcodecode-bytes-storage_deposit_limit-optioncompactu128
+    ContractUploadCode {
+        /// Path to the .wasm artifact
+        #[clap(long, parse(from_os_str))]
+        wasm_path: PathBuf,
+        /// The maximum amount of balance that can be charged/reserved from the caller to pay for the storage consumed
+        #[clap(long)]
+        storage_deposit_limit: Option<u128>,
+    },
+
+    ///  Instantiates a contract from a previously deployed wasm binary.
+    /// API signature: https://polkadot.js.org/docs/substrate/extrinsics/#instantiatevalue-compactu128-gas_limit-compactu64-storage_deposit_limit-optioncompactu128-code_hash-h256-data-bytes-salt-bytes
+    ContractInstantiate {
+        /// balance to transfer from the call origin to the contract
+        #[clap(long, default_value = "0")]
+        balance: u128,
+        /// The gas limit enforced when executing the constructor
+        #[clap(long, default_value = "1000000000")]
+        gas_limit: u64,
+        /// The maximum amount of balance that can be charged/reserved from the caller to pay for the storage consumed
+        #[clap(long)]
+        storage_deposit_limit: Option<u128>,
+        /// Path to the .wasm artifact
+        #[clap(long, parse(from_os_str))]
+        metadata_path: PathBuf,
+        /// Code hash of the deployed contract
+        #[clap(long, parse(try_from_str))]
+        code_hash: H256,
+        /// The name of the contract constructor to call
+        #[clap(name = "constructor", long, default_value = "new")]
+        constructor: String,
+        /// The constructor arguments, encoded as strings
+        #[clap(long, multiple_values = true)]
+        args: Option<Vec<String>>,
+    },
+
+    /// Deploys a new contract, returns its code hash and the AccountId of the instance
+    /// contract cannot already exist on-chain
+    /// API signature: https://polkadot.js.org/docs/substrate/extrinsics/#instantiatewithcodevalue-compactu128-gas_limit-compactu64-storage_deposit_limit-optioncompactu128-code-bytes-data-bytes-salt-bytes
+    ContractInstantiateWithCode {
```

Comment:
This has a lot of overlap with `ContrantInstantiate`. Extract common fields to one struct and use `#[clap(flatten)]`.